### PR TITLE
Move AutoRefresh cache to a separate package, add retries and expose …

### PR DESCRIPTION
**Pull request by EngHabu**
…error on sync

### CODE REVIEW

**Comment by EngHabu** [comment]
PTAL @wild-endeavor, @akhurana001

File: synccache/auto_refresh_cache.go [hunk 1]
```diff
@@ -133,16 +123,56 @@ func (w *autoRefreshCache) sync(ctx context.Context) {
 		// If not ok, it means evicted between the item was evicted between getting the keys and this update loop
 		// which is fine, we can just ignore.
 		if value, ok := w.lruMap.Peek(k); ok {
-			newItem, result, err := w.syncCb(ctx, value.(CacheItem))
+			wrapper := value.(cacheItemWrapper)
+			if wrapper.RetryCount >= w.maxRetries {
```
Comment (wild-endeavor) [inline synccache/auto_refresh_cache.go]: maybe make this >?  Otherwise, if someone specified a max retry of 0, it'll never get synced.

**Comment by wild-endeavor** [comment]
+1 with comment.

File: synccache/auto_refresh_cache.go (hunk 1 above)
Comment (EngHabu) [reply synccache/auto_refresh_cache.go]:
```suggestion
			if wrapper.RetryCount > w.maxRetries {
```

**Pull request closed without merging**